ports/voidstranger/Void Stranger.sh

```diff
@@ -0,0 +1,146 @@
+elif [ -d "$XDG_DATA_HOME/PortMaster/" ]; then
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+source $controlfolder/control.txt
+source $controlfolder/device_info.txt
+export PORT_32BIT="N"
+GAMEDIR="/$directory/ports/voidstranger"
+export LD_LIBRARY_PATH="/usr/lib:/usr/lib32:/$GAMEDIR/lib:$LD_LIBRARY_PATH"
+
+get_controls
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+
+$ESUDO chmod 666 /dev/tty0
+
+cd $GAMEDIR
+
+export GMLOADER_DEPTH_DISABLE=1
+export GMLOADER_SAVEDIR="$GAMEDIR"
+export GMLOADER_PLATFORM="os_linux"
+
+# log the execution of the script into log.txt
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+# here is what we are expecting to get out of the patching process
+expected_chksm111="f38006605356a868087ccff137eb274d"
+
+#first time setup check
+first_time=false
+
+# Check to see if a patched game already exists at the start, and verify with current patch version:
+if [ -f "game.droid" ]; then
+  final_chksm=$(md5sum "game.droid" | awk '{print $1}')
+  if [ "$final_chksm" = "$expected_chksm111" ]; then
+    echo "Found patched game.droid file. Checksum good. md5: ""$final_chksm"
+  else
+    echo "WARNING: game.droid checksum does not match; expecting $expected_chksm; current md5: ""$final_chksm"
+    ##exit 0
+  fi
+else
+  # get data.win checksum
+  game_chksm=$(md5sum gamedata/"data.win" | awk '{print $1}')
+  first_time=true
+  # verify if Steam Version, then patch
+  if [[ -f "gamedata/data.win" ]] && [[ "$game_chksm" = "29f820538024539f18171fb447034fe7" ]]; then
+    echo "Steam Version 1.1.1 Found. Patching data.win. data.win md5 ""$game_chksm"
+    $ESUDO $controlfolder/xdelta3 -d -s gamedata/"data.win" gamedata/"vs.xdelta" gamedata/"vs-patched.win"
+  # or check if it's the Itch version, patch for Steam Version parity and then patch again
+  elif [[ -f "gamedata/data.win" ]] && [[ "$game_chksm" = "1a666b533539af4cebb7c12311bd9a56" ]]; then
+    echo "Itch Version 1.1.1 Found. Patching to be equivalent to Steam version. data.win md5 ""$game_chksm"
+    mv gamedata/"data.win" gamedata/"data_itch.win"
+    $ESUDO $controlfolder/xdelta3 -d -s gamedata/"data_itch.win" gamedata/"vs-itch-to-steam.xdelta" gamedata/"data.win"
+    echo "Patching Updated data.win"
+    $ESUDO $controlfolder/xdelta3 -d -s gamedata/"data.win" gamedata/"vs.xdelta" gamedata/"vs-patched.win"
+    echo "Cleaning up."
+    $ESUDO rm gamedata/"data_itch.win"
+  else
+    echo "Incorrect game checksum or game data not found; check the instructions and your game version. data.win md5 ""$game_chksm"
+    exit 0
+  fi
+  if [ -f "gamedata/vs-patched.win" ]; then 
+    patched_chksm=$(md5sum gamedata/"vs-patched.win" | awk '{print $1}')
+    echo "Patched game checksum expecting $expected_chksm; current md5: ""$patched_chksm"
+    first_time=true
+  fi
+fi
+
+$ESUDO chmod +x "$GAMEDIR/lib/splash"
+
+#SPLASH TIME
```

Comment:
Splash is usually inside the apk file, and will appear if present in apk/assets. Is this a custom splash or a second one to show up before the primary? If not, you can put this in the apk wrapper and ship it with the port.

Reply:
Have you tested this methodology with gmloader-next? Splash function has been removed, so adding the splash.png into the .apk won't generate any splash screen, at least for me with VS. The custom SDL splash is just a workaround for this, but it also adds the "initial loading time" warning for first run users

Reply:
Now that you mention it, maybe it isn't implemented yet...I don't remember.

Reply:
Maybe we could add simple splash back into gmloader-next and merge it with Johnny's repo in the future